File: client-data/src/main/java/com/jraska/github/client/users/data/UsersDataModule.java
package com.jraska.github.client.users.data;

import com.jraska.github.client.PerApp;
import com.jraska.github.client.users.UsersRepository;

import dagger.Module;
import dagger.Provides;
import retrofit2.Retrofit;

@Module
public abstract class UsersDataModule {
  @Provides
  @PerApp
  public static UsersRepository provideUsersRepository(Retrofit retrofit) {
    GitHubUsersApi usersApi = retrofit.create(GitHubUsersApi.class);
    GitHubUserDetailApi detailApi = retrofit.create(GitHubUserDetailApi.class);

    return new GitHubApiUsersRepository(usersApi, detailApi);
  }
}

File: client-data/src/main/java/com/jraska/github/client/users/UsersDataModule.kt
package com.jraska.github.client.users

import com.jraska.github.client.PerApp
import com.jraska.github.client.users.UsersRepository

import dagger.Module
import dagger.Provides
import retrofit2.Retrofit

@Module
object UsersDataModule {
  @JvmStatic
  @Provides
  @PerApp
  fun provideUsersRepository(retrofit: Retrofit): UsersRepository {
    val usersApi = retrofit.create(GitHubUsersApi::class.java)
    val detailApi = retrofit.create(GitHubUserDetailApi::class.java)

    return GitHubApiUsersRepository(usersApi, detailApi)
  }
}
